### PR TITLE
Save the readingMode in Redux

### PR DESCRIPTION
**Pull request by tifa2UP**
### Summary
Refactored the reading mode preference to live in redux.

### Test Plan
Confirmed that the state is now maintained in redux and persists across sessions

### CODE REVIEW

File: src/components/QuranReader/index.tsx [hunk 1]
```diff
@@ -1,15 +1,18 @@
...
-  view: 'translation' | 'reading';
   verses: VerseType[];
 };
 
-const QuranReader = ({ view, verses }: QuranReaderProps) => {
-  if (view === 'reading') {
+const QuranReader = ({ verses }: QuranReaderProps) => {
+  const readingMode = useSelector(selectReadingMode);
+
+  if (readingMode === 'mushaf') {
```
Comment (mmahalwy) [inline src/components/QuranReader/index.tsx:15]: prefer is `'mushaf'` is a constant please 🙏

File: src/redux/slices/QuranReader/readingMode.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+
+export type ReadingMode = 'translation' | 'mushaf';
+
+const initialState: ReadingMode = 'translation';
```
Comment (mmahalwy) [inline src/redux/slices/QuranReader/readingMode.ts:5]: constant please

File: src/components/QuranReader/index.tsx (hunk 1 above)
Comment (tifa2UP) [reply src/components/QuranReader/index.tsx]: Good call, moved it to an enum.

File: src/redux/slices/QuranReader/readingMode.ts (hunk 2 above)
Comment (tifa2UP) [reply src/redux/slices/QuranReader/readingMode.ts]: Good call, moved it to an enum.

**Pull request merged**